addons/mail/static/src/messaging/component/notification_list/notification_list_notification_group_tests.js

```diff
@@ -0,0 +1,569 @@
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationGroup_counter',
+        "should have 1 group counter"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_NotificationGroup_counter').textContent.trim(),
+        "(1)",
+        "should have only 1 notification in the group"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationGroup_date',
+        "should have 1 group date"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_NotificationGroup_date').textContent,
+        "a few seconds ago",
+        "should have the group date corresponding to now"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationGroup_inlineText',
+        "should have 1 group text"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_NotificationGroup_inlineText').textContent.trim(),
+        "An error occurred when sending an email.",
+        "should have the group text corresponding to email"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationGroup_markAsRead',
+        "should have 1 mark as read button"
+    );
+});
+
+QUnit.test('mark as read', async function (assert) {
+    assert.expect(6);
+
+    this.data.initMessaging.mail_failures = [{
+        date: moment.utc().format("YYYY-MM-DD HH:mm:ss"),
+        id: 11,
+        message_type: 'email',
+        model: 'mail.channel',
+        model_name: "Channel",
+        notifications: [{
+            failure_type: 'SMTP',
+            id: 21,
+            notification_status: 'exception',
+            notification_type: 'email',
+            partner_id: [41, "Someone"],
+        }],
+        res_id: 31,
+    }];
+    await this.start({
+        intercepts: {
+            do_action(ev) {
+                assert.step('do_action');
+                assert.strictEqual(
+                    ev.data.action,
+                    'mail.mail_resend_cancel_action',
+                    "action should be the one to cancel email"
+                );
+                assert.strictEqual(
+                    ev.data.options.additional_context.default_model,
+                    'mail.channel',
+                    "action should have the group model as default_model"
+                );
+                assert.strictEqual(
+                    ev.data.options.additional_context.unread_counter,
+                    1,
+                    "action should have the group notification length as unread_counter"
+                );
+            },
+        },
+    });
+    await this.createNotificationListComponent();
+
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationGroup_markAsRead',
+        "should have 1 mark as read button"
+    );
+
+    document.querySelector('.o_NotificationGroup_markAsRead').click();
+    assert.verifySteps(
+        ['do_action'],
+        "should have called do_action correctly"
```

Comment:
again would clarify intent of do_action in comment of such assertions (any other similar assertions should have same treatment)